index.ts

```diff
@@ -603,11 +714,589 @@ class ServerlessCustomDomain {
         }
     }
 
+    /**
+     * Determines whether this plug-in is enabled for a websocket endpoint
+     *
+     * This method reads the customDomain.websockets property "enabled" to see if this plug-in should be enabled.
+     * If the property's value is undefined, a default value of true is assumed (for backwards
+     * compatibility).
+     * If the property's value is provided, this should be boolean, otherwise an exception is thrown.
+     * If no customDomain.websockets object exists, websocket creation is disabled.
+     */
+    public evaluateEnabledWs(): boolean {
```

Comment:
Here's a lot of the reasoning behind why having a list of domain configurations might be better. There's little functional difference in this function as opposed to `evaluateEnabled()` - it just searches for the `enabled` parameter in the `websocket` section. This is repeated throughout the PR, and having a list of domain configs would make the code more DRY.